src/TimeTrace/TimeTraceGenerator.cpp

```diff
@@ -0,0 +1,194 @@
+#include "TimeTraceGenerator.h"
+
+#include <fstream>
+#include <nlohmann\json.hpp>
+
+using namespace Microsoft::Cpp::BuildInsights;
+using namespace Activities;
+using namespace vcperf;
+
+// anonymous namespace to hold auxiliary functions (instead of using it in the class and requiring #include <nlohmann\json.hpp>
+namespace
+{
+
+void AddEntry(const ExecutionHierarchy::Entry* entry, nlohmann::json& traceEvents,
+                const PackedProcessThreadRemapping& remappings, const std::unordered_set<unsigned long long>& ignoredEntries)
+{
+    if (ignoredEntries.find(entry->Id) != ignoredEntries.end()) {
```

Comment:
This shouldn't be necessary once you filter in the Hierarchy builder, right?

Reply:
Yeah, I'll remove it once it's done 👍